src/update-feeds/update-feeds.ts

```diff
@@ -57,32 +58,39 @@ export const startUpdateFeedsLoops = async () => {
     const { dataFeedBatchSize, dataFeedUpdateInterval, providers, contracts } = chain;
+    const dataFeedUpdateIntervalMs = dataFeedUpdateInterval * 1000;
 
     // Create a provider and connect it to the DapiDataRegistry contract.
-    const provider = new ethers.providers.StaticJsonRpcProvider(providers[providerName]);
+    const provider = new ethers.providers.StaticJsonRpcProvider({
+      url: providers[providerName]!.url,
+      timeout: RPC_PROVIDER_TIMEOUT_MS,
+    });
     const dapiDataRegistry = getDapiDataRegistry(contracts.DapiDataRegistry, provider);
 
     logger.debug(`Fetching first batch of dAPIs batches`);
     const firstBatchStartTime = Date.now();
-    const goFirstBatch = await go(async () => {
-      const dapisCountCalldata = dapiDataRegistry.interface.encodeFunctionData('dapisCount');
-      const readDapiWithIndexCalldatas = range(0, dataFeedBatchSize).map((dapiIndex) =>
-        dapiDataRegistry.interface.encodeFunctionData('readDapiWithIndex', [dapiIndex])
-      );
-      const [dapisCountReturndata, ...readDapiWithIndexCallsReturndata] = verifyMulticallResponse(
-        await dapiDataRegistry.callStatic.tryMulticall([dapisCountCalldata, ...readDapiWithIndexCalldatas])
-      );
+    const goFirstBatch = await go(
+      async () => {
+        const dapisCountCalldata = dapiDataRegistry.interface.encodeFunctionData('dapisCount');
+        const readDapiWithIndexCalldatas = range(0, dataFeedBatchSize).map((dapiIndex) =>
+          dapiDataRegistry.interface.encodeFunctionData('readDapiWithIndex', [dapiIndex])
+        );
+        const [dapisCountReturndata, ...readDapiWithIndexCallsReturndata] = verifyMulticallResponse(
+          await dapiDataRegistry.callStatic.tryMulticall([dapisCountCalldata, ...readDapiWithIndexCalldatas])
+        );
 
-      const dapisCount = decodeDapisCountResponse(dapiDataRegistry, dapisCountReturndata!);
-      const firstBatch = readDapiWithIndexCallsReturndata
-        // Because the dapisCount is not known during the multicall, we may ask for non-existent dAPIs. These should be filtered out.
-        .slice(0, dapisCount)
-        .map((dapiReturndata) => ({ ...decodeReadDapiWithIndexResponse(dapiDataRegistry, dapiReturndata), chainId }));
-      return {
-        firstBatch,
-        dapisCount,
-      };
-    });
+        const dapisCount = decodeDapisCountResponse(dapiDataRegistry, dapisCountReturndata!);
+        const firstBatch = readDapiWithIndexCallsReturndata
+          // Because the dapisCount is not known during the multicall, we may ask for non-existent dAPIs. These should be filtered out.
+          .slice(0, dapisCount)
+          .map((dapiReturndata) => ({ ...decodeReadDapiWithIndexResponse(dapiDataRegistry, dapiReturndata), chainId }));
+        return {
+          firstBatch,
+          dapisCount,
+        };
+      },
+      { totalTimeoutMs: dataFeedUpdateIntervalMs, attemptTimeoutMs: dataFeedUpdateIntervalMs }
```

Comment:
You only need to set the `totalTimeoutMs`.

However, when we call `runUpdateFeeds` and fetching the first batch takes `dataFeedUpdateIntervalMs` we should return and not process other batches (because they would overlap with another round of `runUpdateFeeds`).